src/udisks2blockdevices.cpp

```diff
@@ -231,8 +231,18 @@ bool BlockDevices::populated() const
 
 bool BlockDevices::isExternal(const QString &dbusObjectPath)
 {
-    static const QRegularExpression externalBlockDevice(QStringLiteral("^/org/freedesktop/UDisks2/block_devices/%1$").arg(externalDevice));
-    return externalBlockDevice.match(dbusObjectPath).hasMatch();
+    Block *maybeRemoveable = find(dbusObjectPath);
+    if (maybeRemoveable != nullptr) {
+        return maybeRemoveable->isExternal();
+    } else {
+        InterfacePropertyMap interfacePropertyMap;
+        maybeRemoveable = new Block(dbusObjectPath, interfacePropertyMap);
```

Comment:
Sometimes a block device is not there e.g. before it was been put to into m_devices.
In that case we have to create a new block object and check if it's drive has the removeable attribute set to true.
However the object isn't populated. I ideas here? cc: @pvuorela @rainemak

Reply:
Some thoughts without knowing the details too well (so take with a pinch of salt): Is there is some method in the D-Bus proxy object that would refresh those properties? Alternatively, if the issue is that the device gets populated only a bit later when connecting it, then I would think that returning false initially here would be fine, and then once it's populated, the device would be checked again and UI informed about the change.

Reply:
The UDisks2::Block class populates the object in the constructor the waiting for the object to be ready happens all over the place in Udisks2::BlockDevices.
The issue is that before a object when the block devices was checked, then created and populated.
My changes require that it has be done before the object is added to Udisks2::BlockDevices.